src/Mono.Android/Android.Runtime/StreamInvoker.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IO;
+
+namespace Android.Runtime
+{
+	public abstract class StreamInvoker : Stream
+	{
+		const string ExceptionInterceptMessage = "Java exception intercepted";
+
+		protected void WrapJavaCall (Action code)
+		{
+			if (code == null)
+				throw new ArgumentNullException (nameof (code));
+
+			try {
+				code ();
+			} catch (Java.IO.IOException ex) {
```

Comment:
I believe that this should use a [conditional `catch` block](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/when#when-in-a-catch-statement):

```csharp
catch (Java.IO.IOException ex) when (AndroidRuntime.IOExceptionType == BoundStreamIOExceptionType.System) {
    throw new IOException (ExceptionInterceptMessage, ex);
}
```

This will avoid the need for the `throw;` (line 21), and make for an overall "better" "first chance exception" experience, as the full "original" stack trace will be present on the first chance exception, instead of a calls sack with this method as the "top" method.